### PR TITLE
Add query parameter support for fog-google

### PR DESCRIPTION
**Pull request by stanhu**
This requires https://github.com/fog/fog-google/pull/409/files

### CODE REVIEW

**Comment by icco** [comment]
FYI, you'll probably need to update https://github.com/carrierwaveuploader/carrierwave/blob/master/carrierwave.gemspec#L37 to make this work.

**Comment by stanhu** [comment]
@icco Oops, the build is failing here due to https://github.com/fog/fog-google/pull/412/files.

**Comment by stanhu** [comment]
Build failures are unrelated and related to https://github.com/carrierwaveuploader/carrierwave/issues/2333.

**Comment by stanhu** [comment]
@mshibuya Could you take a look here? We're shipping this as a monkey patch in GitLab at the moment.

File: carrierwave.gemspec [hunk 1]
```diff
@@ -34,7 +34,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency "rspec", "~> 3.4"
   s.add_development_dependency "webmock"
   s.add_development_dependency "fog-aws"
-  s.add_development_dependency "fog-google", "<= 0.1.0"
```
[the remark this answers is not in the document]
Comment (mshibuya) [reply carrierwave.gemspec]: How about showing warning message if they try to use `options` parameter support with older version of fog-google?

**Comment by stanhu** [comment]
> May I ask you one more favor, could you make this spec cover fog-google?

Sure, good idea. Done.

**Comment by mshibuya** [comment]
Nice work, thanks!

**Comment by stanhu** [comment]
@mshibuya Would you mind tagging a release now? Thanks!

**Pull request merged**